adafruit_emc2101/emc2101_lut.py

```diff
@@ -82,13 +107,15 @@ def __setitem__(self, index, value):
             del self.lut_values[index]
         elif value > 100.0 or value < 0:
             # Range check
-            raise AttributeError("LUT values must be a fan speed from 0-100%")
+            raise ValueError("LUT values must be a fan speed from 0-100%")
         else:
             self.lut_values[index] = value
-        self._update_lut()
+        if self._defer_update == 0:
+            self._update_lut()
 
     def __repr__(self):
         """return the official string representation of the LUT"""
+        # pylint: disable=consider-using-f-string
```

Comment:
Weird, manually adding `consider-using-f-string` shouldn't trip up the CI, was this causing problems?

Reply:
Yes it did.

I also don't know enough to reformat it as .format() - could you suggest the change please?

Reply:
Here's an example on how to do so from the CircuitPython core API documentation page:

https://docs.circuitpython.org/en/latest/docs/design_guide.html#libraries-examples

Reply:
You can use formatting as well, so it would be:

```python
        return "FanSpeedLUT {:x}".format(id(self))
```